src/components/cpu/cpu-max-usage/cpu-max-usage.js

```diff
@@ -18,110 +18,66 @@ export class CpuMaxUsage extends HTMLElement {
-     */
-    svgSetup () {
-        // Setup the svg element in the DOM
-        this.svg = d3.select('svg')
-        .attr("width", this.width + this.margin.left + this.margin.right)
-        .attr("height", this.height + this.margin.top + this.margin.bottom)
-        .append('g')
-        .attr("transform", "translate(" + this.margin.left + "," + this.margin.top + ")");
-
-        // Create the path to the line in the svg element
-        this.svg.append('path')
-        .attr('d', this.calculateLine()(this.marshalledData))
-        .attr('class', 'line')
-        .style('stroke-width', 2)
-        .style('stroke', 'orange')
-        .style('fill', 'none');
-
-        return this;
+    render () {
+        if (this.graphData && this.defaults) {
+          this.innerHTML = "<line-graph data-margin=" + JSON.stringify(this.defaults.margin) + " data-height=" + this.defaults.height + " data-width=" + this.defaults.width + " data-graph=" + this.graphData + " data-line-color=" + this.defaults.lineColor + "></lineGraph>";
+        }
+      }
+
+      /**
+       * @name dataPoints
+       * @description Sets datapoints this.graphdata
+       * @param {string} data This param is stringified JSON data setting
+       */
+    dataPoints(data){
+        this.graphData = data;
     }
 
     /**
-     * @name axisSetup
+     * @name disconnectedCallback
      * @description
-     * Sets up the X and Y axis for the graph rendered in this component
+     * Call back for when the component is detached from the DOM.
      */
-    axisSetup () {
-        // Scale the graph to the dimensions of the svg
-        var chartProps = {};
-        chartProps['x'] = d3.scaleLinear().range([0, this.width]);
-        chartProps['y'] = d3.scaleLinear().range([this.height, 0]);
-
-        var xAxis = d3.axisBottom(chartProps['x']);
-        var yAxis = d3.axisLeft(chartProps['y']);
-
-        // Add the X Axis
-        this.svg.append("g")
-        .attr("class", "x axis")
-        .attr("transform", "translate(0," + this.height + ")")
-        .call(xAxis);
-
-        // Add the Y Axis
-        this.svg.append("g")
-        .attr("class", "y axis")
-        .call(yAxis);
-
-        return this;
-    }
+    disconnectedCallback() {}
 
     /**
-     * @name dataMarshaller
-     * @description
-     * Marshalls the data before being rendered in the graph
+     * @name observedAttributes
+     * @description Sets what attributes this component will listen for.
+     * @returns {Array} an array of attribute to watch for value changes
      */
-    dataMarshaller () {
-        var value = this.attributes.getNamedItem('data').value;
-        var data = JSON.parse(value);
-        this.marshalledData = [];
-
-        for (let index in data[0].valuesCollection) {
-            this.marshalledData.push({ time: data[0].valuesCollection[index][0], cpu_max_usage: data[0].valuesCollection[index][8] });
-        }
-
-        return this
+    static get observedAttributes() {
```

Comment:
I'm thinking we can have this component make use of the observedAttributes to listen for changes. After some testing this seems to be the route that works best for 2 way databinding frameworks like Angular & React.

Reply:
One thing to keep in mind here is order of operations.  In my testing, which you may have a way around this, I have observed that the observedChangesCallback fires before the connected callback, which has interesting/unintended effects.  Let's sync up on this.

Reply:
after a zoom review everything looks good